lib/schema/string.js

```diff
@@ -577,7 +577,7 @@ SchemaString.prototype.$conditionalHandlers =
       $gte: handleSingle,
       $lt: handleSingle,
       $lte: handleSingle,
-      $options: handleSingle,
```

Comment:
Why is this change necessary? `handleSingle()` will cast it to a string, no?

Reply:
handleSingle() will cast it to a string, but also apply uppercase/lowercase setter to reproduce #7800.
Change to String here like the way `SchemaArray` handles `$options`